File: String Manipulation/AlternatingCharacters.py
#!/bin/python3

import os

# Complete the alternatingCharacters function below.
def alternatingCharacters(s):
    count = 0
    for num in range (len(s)-1):
        if s[num] == s[num+1]:
            count +=1
    return count

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    q = int(input())

    for q_itr in range(q):
        s = input()

        result = alternatingCharacters(s)

        fptr.write(str(result) + '\n')

    fptr.close()

File: String Manipulation/alternating_characters.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {
    static int count;

    // Complete the alternatingCharacters function below.
    static int alternatingCharacters(String s) 
    {
        char a ='A';
        char b= 'B';
        count=0;
        int len = s.length();

        for(int i=0;i<len-1;i++)
        {
            char ch1 = s.charAt(i);
            char ch2 = s.charAt(i+1);

            if(ch1==a && ch2!=b)
            {
                count++;
            }
            else if(ch1==b && ch2!=a)
            {
                count++;
            }
        }

        return count;



    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int q = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int qItr = 0; qItr < q; qItr++) {
            String s = scanner.nextLine();

            int result = alternatingCharacters(s);

            bufferedWriter.write(String.valueOf(result));
            bufferedWriter.newLine();
        }

        bufferedWriter.close();

        scanner.close();
    }
}
